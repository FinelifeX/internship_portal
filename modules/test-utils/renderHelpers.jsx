/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import TR from 'react-test-renderer';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

/**
 * Render React component
 * @param Component
 * @param defaultProps
 * @param props
 * @param children
 * @returns {JSX.Element}
 */
const renderComponent = (Component, { defaultProps, props, children }) => (
  <Component {...defaultProps} {...props}>
    {children}
  </Component>
);

/**
 * Render React component inside MemoryRouter
 * @param Component
 * @param defaultProps
 * @param props
 * @param children
 * @param routerOptions
 * @returns {JSX.Element}
 */
const renderComponentWithRouter = (
  Component,
  { defaultProps, props, children },
  routerOptions,
) => (
  <MemoryRouter {...routerOptions}>
    <Component {...defaultProps} {...props}>
      {children}
    </Component>
  </MemoryRouter>
);

/**
 * Component renderer functions
 * @type {{renderComponent: (function(*, {defaultProps: *, props: *, children: *})), renderComponentWithRouter: (function(*, {defaultProps: *, props: *, children: *}, *))}}
 */
export const ComponentRenderers = {
  renderComponent,
  renderComponentWithRouter,
};

/**
 * Shallow render component using "react-test-renderer"
 * @param Component
 * @param defaultProps
 * @param props
 * @param children
 */
export const shallowRender = (
  Component,
  { defaultProps, props, children } = {},
) => TR.create(renderComponent(Component, { defaultProps, props, children }));

/**
 * Render component using "render()" from "@testing-library/react"
 * @param Component
 * @param defaultProps
 * @param props
 * @param children
 */
export const domRender = (Component, { defaultProps, props, children } = {}) =>
  render(renderComponent(Component, { defaultProps, props, children }));
