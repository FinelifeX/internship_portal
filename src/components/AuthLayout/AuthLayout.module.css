.main {
  min-height: 100%;
  padding: 24px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #f0f2f5;
  overflow: auto;
}

.section {
  max-width: 360px;
  width: 100%;
  display: flex;
  flex-direction: column;
}

@media screen and (max-width: 320px) {
  .section {
    max-width: 300px;
  }
}

.logoContainer {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  margin-bottom: 40px;
}

.logo {
  width: 245px;
}
