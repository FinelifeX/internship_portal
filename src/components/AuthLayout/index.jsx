import React from 'react';
import PT from 'prop-types';
import AkvelonLogo from '../../assets/images/akvelon_logo.png';
import styles from './AuthLayout.module.css';

const AuthLayout = ({ children }) => (
  <main className={styles.main}>
    <section className={styles.section}>
      <div className={styles.logoContainer}>
        <img src={AkvelonLogo} alt="Akvelon Logo" className={styles.logo} />
      </div>
      <div className={styles.content}>{children}</div>
    </section>
  </main>
);

AuthLayout.propTypes = {
  children: PT.element.isRequired,
};

export default AuthLayout;
