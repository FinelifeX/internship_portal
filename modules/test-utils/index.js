import * as RenderHelpers from './renderHelpers';

export { RenderHelpers };
