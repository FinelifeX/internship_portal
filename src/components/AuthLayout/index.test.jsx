import { shallow, render } from 'test-utils/renderHelpers';
import AuthLayout from './index';

describe('AuthLayout component', () => {
  const children = 'Hello, world!';

  it('should render correctly', () => {
    const tree = shallow(render(AuthLayout, { children })).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
