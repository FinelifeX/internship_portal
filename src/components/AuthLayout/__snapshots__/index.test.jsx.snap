// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLayout component should render correctly 1`] = `
<main
  className="main"
>
  <section
    className="section"
  >
    <div
      className="logoContainer"
    >
      <img
        alt="Akvelon Logo"
        className="logo"
        src="test-file-stub"
      />
    </div>
    <div
      className="content"
    >
      Hello, world!
    </div>
  </section>
</main>
`;
